ci/test-coverage.sh

```diff
@@ -1,28 +1,33 @@
 #!/usr/bin/env bash
+
 set -e
-cd "$(dirname "$0")/.."
+here=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
+
+if [[ -z $CI ]]; then
+  echo "This script is used by CI environment. Use scripts/coverage.sh directly if you only want to obtain the coverage report"
+  exit 1
+fi
 
 annotate() {
   ${BUILDKITE:-false} && {
     buildkite-agent annotate "$@"
   }
 }
 
-source ci/upload-ci-artifact.sh
-source scripts/ulimit-n.sh
-
-scripts/coverage.sh "$@"
-
-if [[ -z $CI ]]; then
-  exit
-fi
+# run coverage for all
+"$here/../scripts/coverage.sh"
 
-report=coverage-"${CI_COMMIT:0:9}".tar.gz
-mv target/cov/report.tar.gz "$report"
-upload-ci-artifact "$report"
+# compress coverage reports
+HTML_REPORT_TAR_NAME="coverage-${CI_COMMIT:0:9}.tar.gz"
+HTML_REPORT_TAR_PATH="$here/../$HTML_REPORT_TAR_NAME"
+tar zcf "$HTML_REPORT_TAR_PATH" "$here/../coverage/"
 
+# upload reports to buildkite
+# shellcheck source=ci/upload-ci-artifact.sh
```

Comment:
i don't think `ci/upload-ci-artifact.sh` isn't `shellcheck`-ed by the globbing pattern of the `shellcheck` ci step itself.

maybe, writing explicit `shellcheck source=ci/upload-ci-artifact.sh` is your preference?

Reply:
yeah, I think it's more explicit. I prefer to keep it